controllers/vspherecluster_controller_test.go

```diff
@@ -122,6 +135,19 @@ var _ = Describe("ClusterReconciler", func() {
 				}
 				return len(secret.OwnerReferences) > 0
 			}, timeout).Should(BeTrue())
+
+			By("setting the VSphereCluster's VCenterAvailableCondition to true")
+			Eventually(func() bool {
+				if err := testEnv.Get(ctx, key, instance); err != nil {
+					return false
+				}
+				for _, condition := range instance.GetConditions() {
+					if condition.Type == infrav1.VCenterAvailableCondition {
+						return condition.Status == corev1.ConditionTrue
+					}
+				}
```

Comment:
There is a helper function we borrow from the CAPI codebase `sigs.k8s.io/cluster-api/util/conditions` to get a type of condition from an object, `Has` & `IsTrue`. I would suggest using the same.

Reply:
Updated to use `IsTrue`. It will return false if the condition is not present, so there should be no need for `Has`.